starlite/testing/async_test_client/client.py

```diff
@@ -0,0 +1,213 @@
+import warnings
+from contextlib import AsyncExitStack, contextmanager
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Generator,
+    Generic,
+    Optional,
+    TypeVar,
+    Union,
+)
+
+from anyio.from_thread import BlockingPortal, start_blocking_portal
+
+from starlite import ASGIConnection, HttpMethod, ImproperlyConfiguredException
+from starlite.datastructures import MutableScopeHeaders  # noqa: TC001
+from starlite.exceptions import MissingDependencyException
+from starlite.testing.async_test_client.life_span_handler import LifeSpanHandler
+from starlite.testing.async_test_client.transport import AsyncTestClientTransport
+from starlite.types import AnyIOBackend, ASGIApp, HTTPResponseStartEvent
+
+try:
+    from httpx import USE_CLIENT_DEFAULT, AsyncClient, Response
+except ImportError as e:
+    raise MissingDependencyException(
+        "To use starlite.testing, install starlite with 'testing' extra, e.g. `pip install starlite[testing]`"
+    ) from e
+
+
+if TYPE_CHECKING:
+    from httpx._client import UseClientDefault
+    from httpx._types import (
+        AuthTypes,
+        CookieTypes,
+        HeaderTypes,
+        QueryParamTypes,
+        RequestContent,
+        RequestData,
+        RequestFiles,
+        TimeoutTypes,
+        URLTypes,
+    )
+
+    from starlite.middleware.session.base import BaseBackendConfig, BaseSessionBackend
+
+
+T = TypeVar("T", bound=ASGIApp)
+
+
+def fake_http_send_message(headers: MutableScopeHeaders) -> HTTPResponseStartEvent:
+    headers.setdefault("content-type", "application/text")
+    return HTTPResponseStartEvent(type="http.response.start", status=200, headers=headers.headers)
+
+
+def fake_asgi_connection(app: ASGIApp, cookies: Dict[str, str]) -> ASGIConnection[Any, Any, Any]:
+    scope = {
+        "type": "http",
+        "path": "/",
+        "raw_path": b"/",
+        "root_path": "",
+        "scheme": "http",
+        "query_string": b"",
+        "client": ("testclient", 50000),
+        "server": ("testserver", 80),
+        "method": "GET",
+        "http_version": "1.1",
+        "extensions": {"http.response.template": {}},
+        "app": app,
+        "state": {},
+        "path_params": {},
+        "route_handler": None,
+        "_cookies": cookies,
+    }
+    return ASGIConnection[Any, Any, Any](
+        scope=scope,  # type: ignore[arg-type]
+    )
+
+
+class AsyncTestClient(AsyncClient, Generic[T]):
```

Comment:
So, there is a lot of duplication happening here. Can't we inherit from the sync test client and simply override `exit_stack`?

Reply:
this solves it [149a817](https://github.com/starlite-api/starlite/pull/980/commits/149a817b5cd84819582725c0ada2a9bbdd5d079d)